docs/SETUP_PROVISIONING.md

```diff
@@ -0,0 +1,1505 @@
+that value.
+
+Like GitLab pipelines and variables, Terraform state are specific to a GitLab project, so that's another input
+that is passed as a parameter to the shell script.
+
+Because the script needs to interact with GitLab API, the script needs to know the value for your ``GITLAB_USERNAME``
+and your ``GITLAB_PRIVATE_TOKEN`` variable.
+By default, it assumes they are defined in the ``.env`` file. If it cannot find them there,
+it will prompt the user for value.
+It also needs to know the path to the private ssh key that will allow Ansible to configure the target server, so the user 
+will be prompted for that information.
+
+``ops/scripts/tf_init.sh`` need to be used from the deployment environment directory, an environment specific directory, either ``ops/infrastructure/envs/staging``
+or ``ops/infrastructure/envs/live``
+
+This is how the script is run:
+
+```
+$ cd ops/infrastructure/envs/environment
+$ ../../../scripts/tf_init.sh --project gigascience/forks/rija-gigadb-website --env environment --region region --ssh-key /Users/owner/.ssh/rm-gigadb-sshkey.pem --web-ec2-type t3.micro --bastion-ec2-type t3.micro
+```
+
+where we replace:
+* ``gigascience/forks/rija-gigadb-website`` with the appropriate GitLab project.
+* ``environment`` with ``staging`` or ``live``
+* ``region`` with an AWS region where to deploy the infrastructure
+
+You can type `../../../scripts/tf_init.sh --help` to get the list of all possible arguments.
+
+The script also copies the ``ops/infrastructure/terraform.tf`` root module in the environment-specific directory so 
+that terraform commands can be run from that directory.
+
+##### Deployment environment directories
+
+* For staging: ``cd ops/infractructure/envs/staging``
+* For live: ``cd ops/infractructure/envs/live``
+
+Those directories start empty, but the ``ops/scripts/tf_init.sh`` aforementioned and ``ops/scripts/ansible_init.sh`` below will populate them with necessary files
+so we can run ``terraform`` and ``ansible-playbook`` commands from those directories for a safe provisioning of the desired environment.
+
+In particular, the ``ops/scripts/tf_init.sh`` script will write in a ``.init_vars`` file the answer to the prompted value, so that they are not asked
+again in subsequent runs. ``ops/scripts/ansible_init.sh`` will also source that file.
+
+###### The list of files that will be created or placed in those directories during deployment:
+
+| name | description | used by |
+| --- | --- | --- |
+| ansible.properties | created by ``ansible_init.sh``, holds variables assignment used by Ansible to configure deployed application| ansible-playbook | 
+| getIAMUserNameToJSON.sh | copied by ``ansible_init.sh`` | terraform |
+| output/ | created by ``ansible-playbook`` to store a copy of Docker certs | docker |
+| (webapp|files|users|bastion)_playbook.yml | copied by ``ansible_init.sh`` | ansible-playbook | 
```

Comment:
`(webapp|files|users|bastion)` doesn't seem to be rendered properly in markdown viewer in PHPStorm.

Reply:
Yes, and there was one missing (`bootstrap_playbook.yml`)